File: ClipboardZanager/Tests/ClipboardZanager.Core.Desktop.Tests/TestUtilities.cs
using ClipboardZanager.Core.Desktop.Tests.Mocks;
using ClipboardZanager.Shared.Logs;
using ClipboardZanager.Shared.Services;

namespace ClipboardZanager.Core.Desktop.Tests
{
    public static class TestUtilities
    {
        private static bool _initialized;

        public static void Initialize()
        {
            if (!_initialized)
            {
                Logger.InstanceLogSession = new LogMock();
                Logger.InstanceLogSession.SessionStarted();
                _initialized = true;

                ServiceLocator.SettingProvider = new ServiceSettingProviderMock();
            }
            else
            {
                Logger.InstanceLogSession.SessionStopped();
            }

            GetSettingProvider().ResetSettings();
            ServiceLocator.ResetAll();
        }

        internal static ServiceSettingProviderMock GetSettingProvider()
        {
            if (ServiceLocator.SettingProvider == null || !(ServiceLocator.SettingProvider is ServiceSettingProviderMock))
            {
                ServiceLocator.SettingProvider = new ServiceSettingProviderMock();
            }

            return (ServiceSettingProviderMock)ServiceLocator.SettingProvider;
        }
    }
}

File: Android/app/src/androidTest/java/com/etiennebaudoux/clipboardzanager/TestUtilities.java
package com.etiennebaudoux.clipboardzanager;

import com.etiennebaudoux.clipboardzanager.componentmodel.services.ServiceLocator;
import com.etiennebaudoux.clipboardzanager.mocks.ServiceSettingProviderMock;

public final class TestUtilities {
    private static boolean _initialized;

    public static void initialize() {
        if (!_initialized) {
            _initialized = true;

            ServiceLocator.setSettingProvider(new ServiceSettingProviderMock());
        }

        getSettingProvider().resetSettings();
        ServiceLocator.resetAll();
    }

    public static ServiceSettingProviderMock getSettingProvider() {
        if (ServiceLocator.getSettingProvider() == null || !(ServiceLocator.getSettingProvider() instanceof ServiceSettingProviderMock)) {
            ServiceLocator.setSettingProvider(new ServiceSettingProviderMock());
        }

        return (ServiceSettingProviderMock) ServiceLocator.getSettingProvider();
    }
}
